Backend/ActiveCruzer/BLL/MyRequestBll.cs

```diff
@@ -0,0 +1,159 @@
+﻿using ActiveCruzer.DAL.DataContext;
+using ActiveCruzer.Models;
+using AutoMapper;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ActiveCruzer.BLL
+{
+    public class MyRequestBll : IMyRequestsBll
+    {
+        private readonly IMapper _mapper;
+        private readonly ACDatabaseContext _context;
+
+        /// <summary>
+        /// basic constructur for MyRequests
+        /// </summary>
+        /// <param name="mapper"></param>
+        /// <param name="context"></param>
+        public MyRequestBll(IMapper mapper, ACDatabaseContext context)
+        {
+            _mapper = mapper;
+            _context = context;
+        }
+
+        /// <summary>
+        /// take request with id param and current user id param
+        /// </summary>
+        /// <param name="requestId"></param>
+        /// <param name="userId"></param>
+        /// <returns></returns>
+        public int TakeRequest(int requestId, int userId)
+        {
+            var request = _context.Request.FirstOrDefault(x => x.Id == requestId);
+            if(request != null)
+            {
+                request.Volunteer = userId;
+                request.Status = Request.RequestStatus.Pending;
+                _context.Update(request);
+                _context.SaveChanges();
+                return request.Id;
+            }
+            return 0;
+        }
+
+        /// <summary>
+        /// finish request with request id and user id param
+        /// </summary>
+        /// <param name="requestId"></param>
+        public void FinishRequest(int requestId)
+        {
+            var request = _context.Request.FirstOrDefault(x => x.Id == requestId);
+            if(request != null)
+            {
+                request.Status = Request.RequestStatus.Closed;
+                _context.Update(request);
+                _context.SaveChanges();
+            }
+        }
+
+        /// <summary>
+        /// abort request with user id and request id
+        /// </summary>
+        /// <param name="requestId"></param>
+        /// <param name="userId"></param>
+        public void AbortRequest(int requestId, int userId)
+        {
+            var request = _context.Request.FirstOrDefault(x => x.Id == requestId);
+            if(request != null)
+            {
+                request.Status = Request.RequestStatus.Open;
+                request.Volunteer = null;
```

Comment:
should we really unset the volunteer? Then we do not know anymore who has completed the request

Reply:
yes because this it the abort functionality